framework/db/src/main/java/com/cloud/utils/db/TransactionMBeanImpl.java

```diff
@@ -45,14 +48,15 @@ public void removeTransaction(TransactionLegacy txn) {
 
     @Override
     public int getTransactionCount() {
-        return _txns.size();
+        return totalTransactionCount.intValue();
     }
 
     @Override
     public int[] getActiveTransactionCount() {
-        int[] count = new int[2];
+        int[] count = new int[3];
         count[0] = 0;
         count[1] = 0;
+        count[2] = _txns.size();
```

Comment:
change to `getTransactionCount()`?

Reply:
Removed this, not transactions would return the current size; instead of cumulative/historic size.